File: happynum.py
num=int(input("Enter a number: "))
num1=num
sum1=0
for i in range(1, 100):
    sum = 0
    temp= num

    while temp > 0:
        sum1=sum
        digit = temp% 10
        sum += digit ** 2
        temp //= 10
    print("Sum of squares of digits of", i+1, "th step is", sum)
    if sum == 1:
        print(num1, "is a happy number")
        break
    if sum == sum1:
        print(num1, "is not a happy number")
        break
    num=sum

File: happynum.c
#include <stdio.h>

// Function to calculate sum of squares of digits
/*int sum_of_squares(int n) {
    int sum = 0;
    while (n > 0) {
        int digit = n % 10;
        sum += digit * digit;
        n /= 10;
    }
    return sum;
}*/

int main() {
    int num, i;

    printf("Enter a number: ");
    scanf("%d", &num);

    // Use a for loop to check for happy number (max 100 iterations to avoid infinite loop)
    for (i = 0; i < 100; i++) {
        int sum=0;
        int temp = num;
        // num = sum_of_squares(num);
        while (num > 0) {
        int digit = num % 10;
        sum += digit * digit;
        num /= 10;
        printf("%d ", sum);
        }
        printf("%d ", sum);
        // If it reaches 1, it's a happy number
        if (sum == 1) {
            printf("Happy Number \n");
            break ; // Exit the program
        }

        // If it reaches 4, it's not a happy number (cycle)
        if (sum == 4) {
            printf("Not a Happy Number \n");
            break ; // Exit the program
        }
       
        num=sum;
    }
    return 0;
}


